pkg/eventpersister/persister/persister.go

```diff
@@ -359,9 +364,15 @@ func (p *Persister) marshalEvent(event interface{}, environmentNamespace string)
 }
 
 func (p *Persister) marshalEvaluationEvent(
+	ctx context.Context,
 	e *eventproto.EvaluationEvent,
 	environmentNamespace string,
 ) (string, bool, error) {
+	evaluation, tag := p.convToEvaluation(ctx, e)
+	if err := p.upsertUserEvaluation(ctx, environmentNamespace, tag, evaluation); err != nil {
+		handledCounter.WithLabelValues(codeUpsertUserEvaluationFailed).Inc()
+		return "", true, err
```

Comment:
If it fails to upsert the event, we don't ack the message so it can retry later.